docs/features/asc1/teal/index.md

```diff
@@ -2,7 +2,7 @@ title: The Smart Contract Language
 
 Stateless and stateful smart contracts are written in Transaction Execution Approval Langauge (TEAL). These contracts can be written directly or with Python using the [PyTeal library](pyteal.md).
 
-TEAL is an assembly-like language and is processed with a stack machine. The language is a non-turing-complete language that does not support looping but does support forward branches. TEAL programs are processed one line at a time that push and pop values from the stack. These stack values are either unsigned 64 bit integers or byte strings. TEAL provides a set of operators that operate on the values within the stack. TEAL also allows arguments to be passed into the program from a transaction, a scratch space to temporarily store values for use later in the program, access to grouped or single transaction properties, global values, a couple of pseudo operators, constants and some flow control functions like `bnz` for branching. See [TEAL Specification Reference](../../../reference/teal/specification.md) for more details.
+TEAL is an assembly-like language and is processed by the Algorand Virtual Machine (AVM). The language is a turing-complete language that supports looping and subroutines but limits the amount of time the contract has to execute using a dynamic opcode cost evaluation algorithm.  TEAL programs are processed one line at a time that push and pop values from the stack. These stack values are either unsigned 64 bit integers or byte strings. TEAL provides a set of operators that operate on the values within the stack. TEAL also allows arguments to be passed into the program from a transaction, a scratch space to temporarily store values for use later in the program, access to grouped or single transaction properties, global values, a couple of pseudo operators, constants and flow control functions like `bnz` for branching and `callsub` for calling subroutines. In addition, stateful smart contracts can read and write global storage for the contract and local storage for accounts that opt-in to the contract. See [TEAL Specification Reference](../../../reference/teal/specification.md) for more details.
```

Comment:
capitalize "Turing"

suggestion: "TEAL is a Turing complete language that supports looping and subroutines, but limits..."

suggestion: "TEAL programs are processed one line at a time, pushing and popping values on and off the stack."